mypy/constraints.py

```diff
@@ -274,7 +274,11 @@ def infer_constraints_for_callable(
 
 
 def infer_constraints(
-    template: Type, actual: Type, direction: int, skip_neg_op: bool = False
+    template: Type,
+    actual: Type,
+    direction: int,
+    skip_neg_op: bool = False,
+    can_have_union_overlaping: bool = True,
```

Comment:
Overlapping is spelled with two "p"s; same issue in all occurrences.

Reply:
Oh, yes. You are right. Thank you for the advice! I have fixed it